core/modules/rewrite.cc

```diff
@@ -107,7 +107,7 @@ inline void Rewrite::DoRewrite(bess::PacketBatch *batch) {
 
   next_turn_ = start + cnt;
   if (next_turn_ >= bess::PacketBatch::kMaxBurst) {
-    next_turn_ -= bess::PacketBatch::kMaxBurst;
+    next_turn_ = next_turn_ % num_templates_;
```

Comment:
I seem, I had no idea, thanks for pointing this out.

Do you think it is expensive, even if it's done once per batch?

I can try a lookup table and benchmark the two solution

Alternatively, we can leave the code as it is